pkg/scheduler/framework/v1alpha1/context.go

```diff
@@ -27,7 +27,9 @@ const (
 )
 
 // ContextData is a generic type for arbitrary data stored in PluginContext.
-type ContextData interface{}
+type ContextData interface {
+	Clone() ContextData
```

Comment:
Let's document what the expectations are for the implementations.

For example, for performance reasons, we want to encourage shallow copies of slices or maps, if we only expect to change their elements but not change properties of the elements.

Reply:
Done.

Reply:
I added a comment, we can update it with pointers to concrete example once we have some from  builtin plugins.